src/main/java/com/habicus/core/controller/v1/goal/GoalController.java

```diff
@@ -62,4 +74,25 @@ public void setGoalService(GoalService goalService) {
     LOGGER.info(String.format("Retrieving goals associated with user {}", id));
     return goalService.retrieveGoalsByUserId(id);
   }
+
+  /**
+   * This enables the ability for a particular user to create a {@link Goal} object from the client
+   *
+   * @return
+   */
+  @RequestMapping(
+    value = "/{userId}",
+    method = RequestMethod.POST,
+    consumes = {MediaType.APPLICATION_JSON},
+    produces = MediaType.APPLICATION_JSON
+  )
+  public Response addGoal(@PathVariable String userId, @RequestBody Goal inputGoal) {
+    User user = userService.validateUserById(userId);
```

Comment:
Eventually we'll need to have something here to verify that logged in user is the user making the request.. Otherwise random user could add a goal for another user

Reply:
Yah, I figured for right now since we are bootstrapping, we can add better token auth when sec. is integrated.